bundles/org.openhab.automation.jrubyscripting/src/main/java/org/openhab/binding/jrubyscripting/internal/JRubyScriptEngineFactory.java

```diff
@@ -0,0 +1,115 @@
+import org.osgi.service.component.annotations.Modified;
+
+/**
+ * This is an implementation of a {@link ScriptEngineFactory} for Ruby.
+ * handlers.
+ *
+ * @author Brian O'Connell - Initial contribution
+ */
+@NonNullByDefault
+@Component(service = ScriptEngineFactory.class, configurationPid = "org.openhab.automation.jrubyscripting")
+public class JRubyScriptEngineFactory extends AbstractScriptEngineFactory {
+
+    private final JRubyScriptEngineConfiguration configuration = new JRubyScriptEngineConfiguration();
+
+    // Filter out the File entry to prevent shadowing the Ruby File class which breaks Ruby in spectacularly
+    // difficult ways to debug.
+    private final static Set<String> FILTERED_PRESETS = Set.of("File");
+    private final static Set<String> INSTANCE_PRESETS = Set.of();
+    private final static Set<String> GLOBAL_PRESETS = Set.of("scriptExtension", "automationManager", "ruleRegistry",
+            "items", "voice", "rules", "things", "events", "itemRegistry", "ir", "actions", "se", "audio",
+            "lifecycleTracker");
+
+    private final javax.script.ScriptEngineFactory factory = new org.jruby.embed.jsr223.JRubyEngineFactory();
+
+    // formatter turned off because it does not appropriately format chained streams
+    private final List<String> scriptTypes = Stream
+            .concat(factory.getExtensions().stream(), factory.getMimeTypes().stream())
+            .collect(Collectors.toUnmodifiableList());
+
+    // Adds @ in front of a set of variables so that Ruby recogonizes them as instance variables
+    private static Map.Entry<String, Object> mapInstancePresets(Map.Entry<String, Object> entry) {
+        if (INSTANCE_PRESETS.contains(entry.getKey())) {
+            return Map.entry("@" + entry.getKey(), entry.getValue());
+        } else {
+            return entry;
+        }
+    }
+
+    // Adds $ in front of a set of variables so that Ruby recogonizes them as global variables
+    private static Map.Entry<String, Object> mapGlobalPresets(Map.Entry<String, Object> entry) {
+        if (GLOBAL_PRESETS.contains(entry.getKey())) {
+            return Map.entry("$" + entry.getKey(), entry.getValue());
+        } else {
+            return entry;
+        }
+    }
+
+    // The activate call is activate binding and set the bindings configuration
+    @Activate
+    protected void activate(ComponentContext componentContext, Map<String, Object> config) {
+        configuration.update(config, factory);
+    }
+
+    // The modified call updates configuration for binding
+    @Modified
+    protected void modified(Map<String, Object> config) {
+        configuration.update(config, factory);
+    }
+
+    @Override
+    public List<String> getScriptTypes() {
+        return scriptTypes;
+    }
+
+    @Override
+    public void scopeValues(ScriptEngine scriptEngine, Map<String, Object> scopeValues) {
+
+        // Empty comments prevent the formatter from breaking up the correct streams chaining
```

Comment:
```suggestion
        // Empty comments prevent the formatter from breaking up the correct streams chaining
```

Please also check fix the other static analyis warnings for this.